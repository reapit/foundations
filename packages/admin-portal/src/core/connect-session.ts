import { ReapitConnectBrowserSession } from '@reapit/connect-session'
import Routes from '@/constants/routes'

// Needs to be a singleton as the class is stateful
export const reapitConnectBrowserSession = new ReapitConnectBrowserSession({
  connectClientId: window.reapit.config.cognitoClientId,
  connectOAuthUrl: window.reapit.config.cognitoOAuthUrl,
  connectLoginRedirectPath: Routes.APPROVALS,
})
