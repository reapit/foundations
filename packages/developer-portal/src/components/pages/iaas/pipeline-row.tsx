import React, { FC, useState } from 'react'
import { buildStatusToIntent, buildStatusToReadable, pipelineViewable } from '@/utils/pipeline-helpers'
import Routes from '@/constants/routes'
import {
  Button,
  ButtonGroup,
  StatusIndicator,
  TableCell,
  TableExpandableRow,
  TableExpandableRowTriggerCell,
  TableRow,
  TableRowContainer,
} from '@reapit/elements'
import { useHistory } from 'react-router'
import { navigate, openNewPage } from '../../../utils/navigation'
import { PipelineModelInterface } from '@reapit/foundations-ts-definitions'
import { ApiNames } from '@reapit/utils-common'

export const PipelineRow: FC<{ pipeline: PipelineModelInterface }> = ({ pipeline }) => {
  const history = useHistory()
  const [isOpen, setIsOpen] = useState<boolean>(false)

  return (
    <TableRowContainer>
      <TableRow>
        <TableCell>{pipeline.name}</TableCell>
        <TableCell>
          <StatusIndicator intent={buildStatusToIntent(pipeline.buildStatus as string)} />
          {buildStatusToReadable(pipeline.buildStatus as string)}
        </TableCell>
        <TableCell>{pipeline.appType}</TableCell>
        <TableCell>
          <a target="_blank" href={pipeline.repository} rel="noreferrer">
            {pipeline.repository}
          </a>
        </TableCell>
        <TableExpandableRowTriggerCell
          isOpen={isOpen}
          onClick={() => setIsOpen(!isOpen)}
        ></TableExpandableRowTriggerCell>
      </TableRow>
      <TableExpandableRow isOpen={isOpen}>
        <ButtonGroup>
          <Button
            intent="secondary"
            onClick={navigate(history, Routes.APP_PIPELINE.replace(':appId', pipeline.appId as string))}
          >
            Manage
          </Button>
          {pipelineViewable(pipeline.buildStatus as string) && (
            <Button
              intent="primary"
              onClick={openNewPage(`https://${pipeline.subDomain}${ApiNames(window.reapit.config.appEnv).iaas}`)}
            >
              View
            </Button>
          )}
        </ButtonGroup>
      </TableExpandableRow>
    </TableRowContainer>
  )
}
