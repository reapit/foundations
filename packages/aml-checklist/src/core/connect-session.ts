import { ReapitConnectBrowserSession } from '@reapit/connect-session'
// Referencing config direct rather than via window because window.reapit.config not available at compile time
// and throws a runtime error
import config from '../../config.json'

// Needs to be a singleton as the class is stateful
export const reapitConnectBrowserSession = new ReapitConnectBrowserSession({
  connectClientId: config.cognitoClientId,
  connectOAuthUrl: config.cognitoOAuthUrl,
})
