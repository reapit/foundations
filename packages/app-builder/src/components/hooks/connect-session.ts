import { ReapitConnectBrowserSession } from '@reapit/connect-session'
import React, { useContext } from 'react'

const ConnectSessionContext = React.createContext<ReapitConnectBrowserSession | undefined>(undefined)

export const ConnectSessionProvider = ConnectSessionContext.Provider

export const useConnectSession = () => useContext(ConnectSessionContext)
