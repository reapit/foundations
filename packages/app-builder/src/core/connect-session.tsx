import { ReapitConnectBrowserSession, ReapitConnectBrowserSessionInitializers } from '@reapit/connect-session'

// Needs to be a singleton as the class is stateful
export const reapitConnectBrowserSession = new ReapitConnectBrowserSession({
  connectClientId: window.reapit.config.connectClientId,
  connectOAuthUrl: window.reapit.config.connectOAuthUrl,
  connectUserPoolId: window.reapit.config.connectUserPoolId,
})

const sessions: Record<string, ReapitConnectBrowserSession> = {}

export const getReapitConnectBrowserSession = (config: ReapitConnectBrowserSessionInitializers) => {
  if (!sessions[config.connectClientId]) {
    sessions[config.connectClientId] = new ReapitConnectBrowserSession(config)
  }
  return sessions[config.connectClientId]
}
