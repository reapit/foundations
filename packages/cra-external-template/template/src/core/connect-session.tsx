import { ReapitConnectBrowserSession } from '@reapit/connect-session'
import { Config } from '../types/global'

export const ReapitConnectBrowserSessionInstance = {
  _instance: ReapitConnectBrowserSession,
  get instance() {
    return this._instance
  },
  initInstance: function(config: Config) {
    this._instance = new ReapitConnectBrowserSession({
      connectClientId: config.cognitoClientId,
      connectOAuthUrl: config.cognitoOAuthUrl,
    })
  },
}
