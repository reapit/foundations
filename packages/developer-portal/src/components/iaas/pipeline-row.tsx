import React, { Dispatch, FC, SetStateAction, useState } from 'react'
import { buildStatusToIntent, buildStatusToReadable, pipelineViewable } from '@reapit/utils-common'
import Routes from '../../constants/routes'
import {
  Button,
  ButtonGroup,
  FlexContainer,
  StatusIndicator,
  TableCell,
  TableExpandableRow,
  TableExpandableRowTriggerCell,
  TableRow,
  TableRowContainer,
} from '@reapit/elements'
import { useNavigate } from 'react-router'
import { navigateRoute, openNewPage } from '../../utils/navigation'
import { PipelineModelInterface } from '@reapit/foundations-ts-definitions'
import { ReapitConnectSession } from '@reapit/connect-session'
import { useChannel, useEvent } from '@harelpls/use-pusher'
import { handlePipelineEvent } from '../apps/page/pusher-event-wrapper'

interface PipelineRowProps {
  pipeline: PipelineModelInterface
  connectSession: ReapitConnectSession
}

export const PipelineRow: FC<PipelineRowProps> = ({ pipeline, connectSession }) => {
  const navigate = useNavigate()
  const [appPipeline, setAppPipeline] = useState<PipelineModelInterface>(pipeline)
  const [isOpen, setIsOpen] = useState<boolean>(false)

  const channel = useChannel(`private-${connectSession?.loginIdentity.developerId}`)
  useEvent(
    channel,
    'pipeline-update',
    handlePipelineEvent(
      appPipeline,
      setAppPipeline as Dispatch<SetStateAction<PipelineModelInterface | null>>,
      pipeline.id ?? null,
    ),
  )

  return (
    <TableRowContainer>
      <TableRow>
        <TableCell>{appPipeline.name}</TableCell>
        <TableCell>
          <StatusIndicator intent={buildStatusToIntent(appPipeline.buildStatus as string)} />
          {buildStatusToReadable(appPipeline.buildStatus as string)}
        </TableCell>
        <TableCell>
          <a target="_blank" href={appPipeline.repository} rel="noreferrer">
            {appPipeline.repository}
          </a>
        </TableCell>
        <TableExpandableRowTriggerCell
          isOpen={isOpen}
          onClick={() => setIsOpen(!isOpen)}
        ></TableExpandableRowTriggerCell>
      </TableRow>
      <TableExpandableRow isOpen={isOpen}>
        <FlexContainer isFlexAlignCenter isFlexJustifyCenter>
          <ButtonGroup>
            <Button
              intent="primary"
              onClick={navigateRoute(
                navigate,
                Routes.APP_PIPELINE_CONFIGURE.replace(':appId', appPipeline.appId as string),
              )}
            >
              Configure
            </Button>
            <Button
              intent="primary"
              onClick={navigateRoute(navigate, Routes.APP_PIPELINE.replace(':appId', appPipeline.appId as string))}
            >
              Deployments
            </Button>
            {pipelineViewable(appPipeline.buildStatus as string) && (
              <Button intent="default" onClick={openNewPage(`https://${appPipeline.subDomain}.iaas.paas.reapit.cloud`)}>
                View App
              </Button>
            )}
          </ButtonGroup>
        </FlexContainer>
      </TableExpandableRow>
    </TableRowContainer>
  )
}
