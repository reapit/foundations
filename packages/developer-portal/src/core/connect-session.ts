import { ReapitConnectBrowserSession } from '@reapit/connect-session'

// Needs to be a singleton as the class is stateful
export const reapitConnectBrowserSession = new ReapitConnectBrowserSession({
  connectClientId: window.reapit.config.cognitoClientId,
  connectOAuthUrl: window.reapit.config.cognitoOAuthUrl,
  connectLoginRedirectPath: '/apps',
  connectUserPoolId: window.reapit.config.cognitoUserPoolId,
})
