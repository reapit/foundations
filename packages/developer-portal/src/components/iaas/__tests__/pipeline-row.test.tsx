import { ReapitConnectSession } from '@reapit/connect-session'
import React from 'react'
import { render } from '../../../tests/react-testing'
import { PipelineRow } from '../pipeline-row'

describe('PipelineRow', () => {
  it('should match snapshot', () => {
    expect(
      render(
        <PipelineRow
          connectSession={
            {
              loginIdentity: {
                developerId: '',
              },
            } as ReapitConnectSession
          }
          pipeline={{
            buildStatus: 'IN_PROGRESS',
          }}
        />,
      ),
    ).toMatchSnapshot()
  })
})
