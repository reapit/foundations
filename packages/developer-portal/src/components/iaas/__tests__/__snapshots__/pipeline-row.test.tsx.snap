// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelineRow should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-93 el-snack-holder"
      />
      <div
        class="mocked-styled-110 el-table-row-container"
      >
        <div
          class="mocked-styled-103 el-table-row"
        >
          <div
            class="el-table-cell-has-dark-text el-table-cell"
          >
            <div
              class="mocked-styled-111 el-table-cell-content"
            />
          </div>
          <div
            class="el-table-cell-has-dark-text el-table-cell"
          >
            <div
              class="mocked-styled-111 el-table-cell-content"
            >
              <span
                class="el-intent-neutral el-status-indicator"
              />
              In Progress
            </div>
          </div>
          <div
            class="el-table-cell-has-dark-text el-table-cell"
          >
            <div
              class="mocked-styled-111 el-table-cell-content"
            >
              <a
                rel="noreferrer"
                target="_blank"
              />
            </div>
          </div>
          <div
            class="mocked-styled-109 el-table-expandable-row-trigger-cell"
            role="button"
            tabindex="0"
          >
            <span
              class="el-intent-default el-icon-new el-icon"
            >
              <svg
                fill="none"
                height="1em"
                role="img"
                title="Icon image with name more"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 12a2 2 0 1 1-4 0 2 2 0 0 1 4 0M12 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4M19 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="mocked-styled-113 el-table-expandable-row"
        >
          <div
            class="mocked-styled-114 el-table-expandable-container"
          >
            <div
              class="el-flex-container el-flex-justify-center el-flex-align-center"
            >
              <div
                class="mocked-styled-2 el-button-group"
              >
                <div
                  class="mocked-styled-3 el-button-group-inner"
                >
                  <button
                    class="el-intent-primary el-button"
                  >
                    <div
                      class="mocked-styled-0 el-button-loader"
                    />
                    Configure
                  </button>
                  <button
                    class="el-intent-primary el-button"
                  >
                    <div
                      class="mocked-styled-0 el-button-loader"
                    />
                    Deployments
                  </button>
                  <button
                    class="el-intent-neutral el-button"
                  >
                    <div
                      class="mocked-styled-0 el-button-loader"
                    />
                    View App
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-93 el-snack-holder"
    />
    <div
      class="mocked-styled-110 el-table-row-container"
    >
      <div
        class="mocked-styled-103 el-table-row"
      >
        <div
          class="el-table-cell-has-dark-text el-table-cell"
        >
          <div
            class="mocked-styled-111 el-table-cell-content"
          />
        </div>
        <div
          class="el-table-cell-has-dark-text el-table-cell"
        >
          <div
            class="mocked-styled-111 el-table-cell-content"
          >
            <span
              class="el-intent-neutral el-status-indicator"
            />
            In Progress
          </div>
        </div>
        <div
          class="el-table-cell-has-dark-text el-table-cell"
        >
          <div
            class="mocked-styled-111 el-table-cell-content"
          >
            <a
              rel="noreferrer"
              target="_blank"
            />
          </div>
        </div>
        <div
          class="mocked-styled-109 el-table-expandable-row-trigger-cell"
          role="button"
          tabindex="0"
        >
          <span
            class="el-intent-default el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name more"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 12a2 2 0 1 1-4 0 2 2 0 0 1 4 0M12 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4M19 14a2 2 0 1 0 0-4 2 2 0 0 0 0 4"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="mocked-styled-113 el-table-expandable-row"
      >
        <div
          class="mocked-styled-114 el-table-expandable-container"
        >
          <div
            class="el-flex-container el-flex-justify-center el-flex-align-center"
          >
            <div
              class="mocked-styled-2 el-button-group"
            >
              <div
                class="mocked-styled-3 el-button-group-inner"
              >
                <button
                  class="el-intent-primary el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Configure
                </button>
                <button
                  class="el-intent-primary el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Deployments
                </button>
                <button
                  class="el-intent-neutral el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  View App
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
